gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/EncounteMapperTest.java

```diff
@@ -0,0 +1,334 @@
+    private static final String CODING_DISPLAY = "Ischaemic heart disease";
+    private static final String FULL_VALID_STRUCTURED_ENCOUNTER_XML = "full_valid_structured_encounter.xml";
+    private static final String FULL_VALID_STRUCTURED_ENCOUNTER_WITH_LINKSET_XML = "full_valid_structured_encounter_with_linkset.xml";
+    private static final String FULL_VALID_FLAT_ENCOUNTER_XML = "full_valid_flat_encounter.xml";
+    private static final String NO_OPTIONAL_FLAT_ENCOUNTER_XML = "no_optional_valid_flat_encounter.xml";
+    private static final String INVALID_ENCOUNTER_CODE_1_XML = "invalid_encounter_code_1.xml";
+    private static final String INVALID_ENCOUNTER_CODE_2_XML = "invalid_encounter_code_2.xml";
+    private static final String SUPPRESSED_COMPOSITION_WITH_EHR_EMPTY_XML = "suppressed_with_ehr_empty.xml";
+    private static final String SUPPRESSED_COMPOSITION_WITH_REGISTRATION_STATEMENT_XML = "suppressed_with_ehr_registration.xml";
+    private static final String EFFECTIVE_CENTER_ENCOUNTER_PERIOD_XML = "effective_center_encounter_period.xml";
+    private static final String EFFECTIVE_LOW_AND_HIGH_ENCOUNTER_PERIOD_XML = "effective_low_and_high_encounter_period.xml";
+    private static final String EFFECTIVE_LOW_ENCOUNTER_PERIOD_XML = "effective_low_encounter_period.xml";
+    private static final String EFFECTIVE_HIGH_ENCOUNTER_PERIOD_XML = "effective_high_encounter_period.xml";
+    private static final String EFFECTIVE_CENTER_NULL_FLAVOR_ENCOUNTER_PERIOD_XML = "effective_center_null_flavor_encounter_period.xml";
+    private static final String AVAILABILITY_TIME_ENCOUNTER_PERIOD_XML = "availability_time_encounter_period.xml";
+    private static final String NO_ENCOUNTER_PERIOD_XML = "no_encounter_period.xml";
+    private static final String ENCOUNTER_WITH_MULTIPLE_COMPOUND_STATEMENTS_XML = "encounter_with_multiple_compound_statements.xml";
+
+    @MockBean
+    private IdGeneratorService idGenerator;
+
+    @Autowired
+    private EncounterMapper encounterMapper;
+
+    @Mock
+    private CodeableConceptMapper codeableConceptMapper;
+
+    private Patient patient;
+
+    @BeforeEach
+    public void setup() {
+        patient = new Patient();
+        patient.setId(PATIENT_ID);
+        setUpCodeableConceptMock();
+        when(idGenerator.generateUuid()).thenReturn(FLAT_TOPIC_ID);
+    }
+
+    @Test
+    public void testEncountersWithMultipleCompoundStatements() {
+        var ehrExtract = unmarshallEhrExtractElement(ENCOUNTER_WITH_MULTIPLE_COMPOUND_STATEMENTS_XML);
+
+        Map<String, List<? extends DomainResource>> mappedResources = encounterMapper.mapEncounters(ehrExtract, patient);
+
+        var encounterList = mappedResources.get(ENCOUNTER_KEY);
+        assertThat(mappedResources.get(ENCOUNTER_KEY).size()).isOne();
+        assertThat(mappedResources.get(CONSULTATION_KEY).size()).isOne();
+        assertThat(mappedResources.get(TOPIC_KEY).size()).isEqualTo(2);
```

Comment:
The style checks will want these "magic numbers" defined as constants

Reply:
Strangely i think it still passes with 2. I've added a constant regardless.